cmd/flux/check_test.go

```diff
@@ -0,0 +1,40 @@
+// +build e2e
+
+package main
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/fluxcd/flux2/internal/utils"
+)
+
+func TestCheckPre(t *testing.T) {
+	out, err := utils.ExecKubectlCommand(context.TODO(), utils.ModeCapture, rootArgs.kubeconfig, rootArgs.kubecontext, "version", "--short")
```

Comment:
Please use `kubectl version --output json` then use the json objects to extract the versions.